### PR TITLE
Data products table and index structure updates, and adding some unit test documentation

### PR DESCRIPTION
**Pull request by rizeladiaz**
I changed the data products table to include input and output files, units and missing product files. Changed the left-hand side index order.
Added some text to the file_naming page 
Added a new page to document all AMI tests. This is as a sample for what we need to do fo the rest of the steps.

### CODE REVIEW

File: docs/jwst/data_products/file_naming.rst [hunk 1]
```diff
@@ -28,8 +30,16 @@ An example Stage 1 product FITS file name is::
...
 
-Stage 3 Naming
-^^^^^^^^^^^^^^
+Association file names
+^^^^^^^^^^^^^^^^^^^^^^
+Associations capture the relationship between exposures that are to be combined by design to 
+form a single product. Association data products result from the combination of multiple exposures like dithers 
+or mosaics and are generated by the stage 3 of the calibration pipeline. 
+
+The format for the file name of an association has all alphabetic characters in lower case, underscores
```
Comment (hbushouse) [inline docs/jwst/data_products/file_naming.rst]: "... of an association **product** has all alphabetic ..."

File: docs/jwst/data_products/product_types.rst [hunk 2]
```diff
@@ -1,81 +1,131 @@
 Data Product Types
 ------------------
-The following table contains a list of all data product types, as given by their file name suffix,
-the pipeline and stage of processing that creates each type, and an indication of whether the file naming
-is exposure-based or source-based. The product name suffixes are active links to detailed descriptions
-in the following sections.
+The following tables contain a list of all data product types, as given by their file name suffix. There is one table per stage of processing.
```
Comment (hbushouse) [inline docs/jwst/data_products/product_types.rst]: "tables" is plural, hence "list" also needs to be plural: The following tables contain lists of all data product ..."

File: docs/jwst/data_products/product_types.rst [hunk 3]
```diff
@@ -1,81 +1,131 @@
...
+All tables indicate whether the file naming is exposure-based (Exp) or source-based (Srce).
```
Comment (hbushouse) [inline docs/jwst/data_products/product_types.rst]: I would drop the "e" from the end of "Srce". It just looks funny (at least to me) and most astronomers understand the abbreviation "src" (it also saves one letter's worth of space in the table).

File: docs/jwst/data_products/product_types.rst [hunk 4]
```diff
@@ -1,81 +1,131 @@
...
+When the product is not created by default, the flag *Optional* is indicated in the
+description. The different stages of the calibration pipeline and are as defined in
```
Comment (hbushouse) [inline docs/jwst/data_products/product_types.rst]: Fix wording: "The different stages of the calibration pipeline are defined in ..."

File: docs/jwst/ami/ami_test.rst [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+test_ami_analyze
+----------------
+
+utils module tests:
++++++++++++++++++++
+
+For the module *utils* we have several tests that compare the calculated value with a known value. The tests are:
+
+ - Test of rebin() and krebin() in utils module
```
[the remark this answers is not in the document]
Comment (dmggh) [reply docs/jwst/ami/ami_test.rst]: Howard's comment just above also applies to the other modules (leastsqrm, hexee, etc)

File: docs/jwst/ami/ami_test.rst [hunk 6]
```diff
@@ -0,0 +1,105 @@
+.. _ami_unit_test:
+
+ami unit tests
```
[the remark this answers is not in the document]
Comment (rizeladiaz) [reply docs/jwst/ami/ami_test.rst]: I have done that before but looks odd given that the other sections have the name of the step in lowercase. In any case, I made the change to uppercase as requested.

File: docs/jwst/ami/ami_test.rst [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+==============
+
+There are unit test for AMI Analyze and AMI interfarce.
```
[the remark this answers is not in the document]
Comment (rizeladiaz) [reply docs/jwst/ami/ami_test.rst]: Added the "s" to test and corrected the typo

File: docs/jwst/ami/ami_test.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (rizeladiaz) [reply docs/jwst/ami/ami_test.rst]: I removed the "... in .... module ..." part for all the modules

File: docs/index.rst [hunk 8]
```diff
@@ -14,16 +14,6 @@ User Manual
...
-=====================
-
-.. toctree::
-   :maxdepth: 1
-
-   jwst/package_index.rst
-
-
 ==========================
 Dataproducts Documentation
```
[the remark this answers is not in the document]
Comment (rizeladiaz) [reply docs/index.rst]: Done

File: docs/jwst/ami/main.rst [hunk 9]
```diff
@@ -12,3 +12,5 @@ of three steps:
 These three steps are applied to an association of AMI exposures using the
 pipeline module :ref:`calwebb_ami3 <calwebb_ami3>`.
 
+Unit test for this steps can be found in the :ref:`AMI unit test information <ami_unit_test>`
```
[the remark this answers is not in the document]
Comment (rizeladiaz) [reply docs/jwst/ami/main.rst]: Done

File: docs/jwst/data_products/file_naming.rst [hunk 10]
```diff
@@ -1,11 +1,13 @@
...
 -------------------
 
 .. _exp_file_names:
 
-Stage 0 - 2 Naming
-^^^^^^^^^^^^^^^^^^
-The FITS file naming scheme for Stage 0, 1, and 2 "exposure-based" products is::
+Exposure file names
+^^^^^^^^^^^^^^^^^^^
+The names of the exposure level data (stage 0 to 2) is constructed with information from the science header of the exposure, allowing users to map it to the observation in ther corresponding APT files. For Stage 0, 1, and 2 "exposure-based" products is:
```
[the remark this answers is not in the document]
Comment (rizeladiaz) [reply docs/jwst/data_products/file_naming.rst]: Done

File: docs/jwst/ami/main.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (rizeladiaz) [reply docs/jwst/ami/main.rst]: Done

File: docs/jwst/data_products/file_naming.rst [hunk 11]
```diff
@@ -28,8 +30,16 @@ An example Stage 1 product FITS file name is::
 
 .. _src_file_names:
 
-Stage 3 Naming
-^^^^^^^^^^^^^^
+Association file names
```
[the remark this answers is not in the document]
Comment (rizeladiaz) [reply docs/jwst/data_products/file_naming.rst]: I changed the title to "Stage 3 file names". I also changed the text in the first paragraph to clarify how these Stage 3 products relate to an association.

File: docs/jwst/data_products/file_naming.rst (hunk 1 above)
Comment (rizeladiaz) [reply docs/jwst/data_products/file_naming.rst]: changed it to "...Stage 3 association product..."

File: docs/jwst/data_products/product_types.rst (hunk 2 above)
Comment (rizeladiaz) [reply docs/jwst/data_products/product_types.rst]: Done

File: docs/jwst/data_products/product_types.rst (hunk 3 above)
Comment (rizeladiaz) [reply docs/jwst/data_products/product_types.rst]: Done

File: docs/jwst/data_products/product_types.rst (hunk 4 above)
Comment (rizeladiaz) [reply docs/jwst/data_products/product_types.rst]: Done

**Pull request merged**